File: bootstrap/src/main/java/net/dontdrinkandroot/wicket/bootstrap/component/form/formgroup/FormGroupInputUrl.java
package net.dontdrinkandroot.wicket.bootstrap.component.form.formgroup;

import org.apache.wicket.Component;
import org.apache.wicket.markup.html.form.UrlTextField;
import org.apache.wicket.model.IModel;

import net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.InputGroupUrl;


public class FormGroupInputUrl extends FormGroupInputGroup<String, String, UrlTextField, InputGroupUrl>
{

	public FormGroupInputUrl(String id, IModel<String> labelModel, IModel<String> model)
	{
		super(id, labelModel, model);
	}

	@Override
	protected InputGroupUrl createInputGroup(String id)
	{
		return new InputGroupUrl(id, this.getModel()) {

			@Override
			protected Component createInputGroupAddonBefore(String id)
			{
				return FormGroupInputUrl.this.createInputGroupAddonBefore(id);
			}

			@Override
			protected Component createInputGroupAddonAfter(String id)
			{
				return FormGroupInputUrl.this.createInputGroupAddonAfter(id);
			}
		};
	}
}

File: bootstrap/src/main/kotlin/net/dontdrinkandroot/wicket/bootstrap/component/form/formgroup/FormGroupInputUrl.kt
package net.dontdrinkandroot.wicket.bootstrap.component.form.formgroup

import net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.InputGroupUrl
import org.apache.wicket.Component
import org.apache.wicket.markup.html.form.UrlTextField
import org.apache.wicket.model.IModel

class FormGroupInputUrl(id: String, model: IModel<String>, labelModel: IModel<String>) :
    FormGroupInputGroup<String, String, UrlTextField, InputGroupUrl>(id, model, labelModel) {

    override fun createInputGroup(id: String): InputGroupUrl {
        return object : InputGroupUrl(id, model) {
            override fun createAddonPrepend(id: String): Component {
                return this@FormGroupInputUrl.createInputGroupPrepend(id)
            }

            override fun createAddonAppend(id: String): Component {
                return this@FormGroupInputUrl.createInputGroupAppend(id)
            }
        }
    }
}